lib/Sema/TypeCheckType.cpp

```diff
@@ -305,9 +305,11 @@ Type TypeChecker::resolveTypeInContext(
       continue;
 
     // Search the type of this context and its supertypes.
-    for (auto fromType = resolver->resolveTypeOfContext(parentDC);
-         fromType;
-         fromType = getSuperClassOf(fromType)) {
+    auto fromType = resolver->resolveTypeOfContext(parentDC);
+    for (;;) {
```

Comment:
I don’t understand why it was necessary to rewrite the `for` loop this way.  Is there a reason?